cirq/google/calibration/engine_simulator.py

```diff
@@ -0,0 +1,405 @@
+            assert np.isclose(gate.theta, np.pi / 4) and np.isclose(
+                gate.phi, 0.0
+            ), f'Expected ISWAP ** -0.5 like gate, got {gate}'
+            return PhasedFSimCharacterization(
+                theta=np.pi / 4, zeta=0.0, chi=0.0, gamma=0.0, phi=0.0
+            )
+
+        if simulator is None:
+            simulator = Simulator()
+
+        return PhasedFSimEngineSimulator(
+            simulator, drift_generator=sample_gate, gates_translator=try_convert_sqrt_iswap_to_fsim
+        )
+
+    @staticmethod
+    def create_with_random_gaussian_sqrt_iswap(
+        mean: PhasedFSimCharacterization,
+        *,
+        simulator: Optional[Simulator] = None,
+        sigma: PhasedFSimCharacterization = PhasedFSimCharacterization(
+            theta=0.02, zeta=0.05, chi=0.05, gamma=0.05, phi=0.02
+        ),
+        random_or_seed: RANDOM_STATE_OR_SEED_LIKE = None,
+    ) -> 'PhasedFSimEngineSimulator':
+        """Creates a PhasedFSimEngineSimulator that introduces a random deviation from the mean.
+
+        The random deviations are described by a Gaussian distribution of a given mean and sigma,
+        for each angle respectively.
+
+        Each gate for each pair of qubits retains the sampled values for the entire simulation, even
+        weh used multiple times within a circuit.
+
+        Attributes:
+            mean: The mean value for each unitary angle. All parameters must be provided.
+            simulator: Simulator object to use. When None, a new instance of cirq.Simulator() will
+                be created.
+            sigma: The standard deviation for each unitary angle. When some sigma parameter is None,
+                the mean value will be always used for that parameter.
+
+        Returns:
+            New PhasedFSimEngineSimulator instance.
+        """
+
+        def sample_gate(_1: Qid, _2: Qid, gate: FSimGate) -> PhasedFSimCharacterization:
+            assert np.isclose(gate.theta, np.pi / 4) and np.isclose(
+                gate.phi, 0.0
+            ), f'Expected ISWAP ** -0.5 like gate, got {gate}'
+
+            def sample_value(
+                gaussian_mean: Optional[float], gaussian_sigma: Optional[float]
+            ) -> float:
+                assert gaussian_mean is not None
+                if gaussian_sigma is None:
+                    return gaussian_mean
+                return rand.normal(gaussian_mean, gaussian_sigma)
+
+            return PhasedFSimCharacterization(
+                theta=sample_value(mean.theta, sigma.theta),
+                zeta=sample_value(mean.zeta, sigma.zeta),
+                chi=sample_value(mean.chi, sigma.chi),
+                gamma=sample_value(mean.gamma, sigma.gamma),
+                phi=sample_value(mean.phi, sigma.phi),
+            )
+
+        if mean.any_none():
+            raise ValueError(f'All mean values must be provided, got mean of {mean}')
+
+        rand = parse_random_state(random_or_seed)
```

Comment:
I get weirded out when variables used in closures are defined after the closure, although you've taught me it works fine. I still think it would read better if the `parse_random_state` and mean check goes above the definition of `sample_gate`

Reply:
Heh, done.